### PR TITLE
Batfishification: build illustrations in the vector tile spec

### PR DESCRIPTION
**Pull request by colleenmcginnis**
Ref #75

This PR builds out functionality for interactive/animated illustrations on the vector tile specification page:

- [x] `EncodedGeometryDiagram` (see [current implementation](https://docs.mapbox.com/vector-tiles/specification/#encoding-geom))
  - [x] first draft (https://github.com/mapbox/vector-tiles/pull/77/commits/912c082876a9fddd1a36454bd4e1b0127ae9bde7)
  - [x] ~~[help needed] get only one "pen" per step~~ not essential
  - [x] [help needed] grid canvas to reset when restarting at step 1
  - [x] [help needed] general design / layout feedback
- [x] `EncodedAttributesDiagram` (see [current implementation](https://docs.mapbox.com/vector-tiles/specification/#encoding-attr)) (https://github.com/mapbox/vector-tiles/pull/77/commits/f53dc75a9e59e1f929c206714434097a3da87f97 https://github.com/mapbox/vector-tiles/pull/77/commits/7cba81388e17404f0eb11542cae799b0c2dc756b)
- [x] `WindingOrderDiagram` (see [current implementation](https://docs.mapbox.com/vector-tiles/specification/#winding-order)) (https://github.com/mapbox/vector-tiles/pull/77/commits/1d849fd4550031552c3a7d8213cc207b52ea207c)
- [x] `NotIncludedDiagram` (see [current implementation](https://docs.mapbox.com/vector-tiles/specification/#simplification))
  - [x] first draft (https://github.com/mapbox/vector-tiles/pull/77/commits/05af57aedb827015e9062a6bdc2a7da0d61a6401)
  - [x] ~~[help needed] fading on step change~~ not essential

cc @mapsam

### CODE REVIEW

File: src/data/encoding-geometry-steps.js [hunk 1]
```diff
@@ -0,0 +1,216 @@
+import md from '@mapbox/batfish/modules/md';
+
+export const encodingGeometrySteps = [
+  {
```
Comment (katydecorah) [inline src/data/encoding-geometry-steps.js]: Should there be a Step 0 here?

image

File: src/components/illustrations/encoding-geometry-diagram.js [hunk 2]
```diff
@@ -1,15 +1,97 @@
...
+                    stepIndex:
+                      this.state.stepIndex < encodingGeometrySteps.length - 1
+                        ? this.state.stepIndex + 1
+                        : 0
+                  });
+                }}
+                corners={true}
+              >
+                {this.state.stepIndex === encodingGeometrySteps.length - 1
+                  ? 'Restart'
```
Comment (katydecorah) [inline src/components/illustrations/encoding-geometry-diagram.js]: The restart button doesn't appear to reset the board

File: src/data/encoding-geometry-steps.js [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+    command: function() {},
+    pen: function(cmd) {
+      cmd.pen(4, 2);
+    }
+  },
+  {
+    name: 'Step 7',
+    color: '#ee4e8b',
+    description: md`
+Since the <span className='bg-pink color-white txt-bold px6 pb6 pt3 round txt-s'>pink polygon</span> is technically a \"hole\" of the <span className='bg-blue color-white txt-bold px6 pb6 pt3 round txt-s'>blue polygon</span> it is considered an <em>interior ring</em>. The Mapbox Vector Tile Specification requires this interior ring to be drawn counter-clockwise, opposite of the <span className='bg-blue color-white txt-bold px6 pb6 pt3 round inline-block txt-s'>blue polygon</span>.
```
[the remark this answers is not in the document]
Comment (colleenmcginnis) [reply src/data/encoding-geometry-steps.js]: Oh wow. This is beautiful. 💅

**Comment by colleenmcginnis** [comment]
Thanks @katydecorah! This is ready for another look.

👋 @mapsam if you'd like to take a look now that's 🆒 if you want to wait until next week when I put it up on staging that's also 🆒.

**Comment by katydecorah** [comment]
Looks fab @colleenmcginnis!

**Pull request merged**